tests/helpers/test-errors-on-directive-linking.js

```diff
@@ -0,0 +1,122 @@
+/* jshint es3: false, esnext: true */
+/* global jasmine, beforeEach, describe, it, expect, module, inject */
+
+(function () {
+
+    var helpers = window._helpers = (window._helpers || {});
+
+    helpers.testErrorsOnDirectiveLinking = function (moduleName, options) {
+
+        'use strict';
+
+        if (!options) {
+            throw new Error('options required');
+        }
+
+        if (!options.directives) {
+            throw new Error('directives option required');
+        }
+        if (!options.factory) {
+            throw new Error('factory options required');
+        }
+
+        var factoryName = options.factory.name;
+        var factoryMethods = options.factory.methods;
+
+
+        describe(moduleName + ': getting fields from OPTIONS request', function () {
```

Comment:
We have previously found (in Python tests) that too much automation in the tests results in issues debugging the tests and understanding them. The result has been that we try to make tests as simple as possible even if involves them not being DRY.